source/grid/grid_generator.cc

```diff
@@ -28,11 +29,1219 @@
+             const unsigned int number_points,
+             const unsigned int factor)
+        {
+          // number of non_equidistant airfoilpoints among which will be
+          // interpolated
+          const unsigned int n_airfoilpoints = factor * number_points;
+
+          // create equidistant airfoil points for upper and lower side
+          return {{make_points_equidistant(
+                     naca_create_points(serialnumber, n_airfoilpoints, true),
+                     number_points),
+                   make_points_equidistant(
+                     naca_create_points(serialnumber, n_airfoilpoints, false),
+                     number_points)}};
+        }
+
+        /**
+         * create(number_points)-NACA points for either upper or lower side
+         * calls function naca_create_points_4_digits()
+         *
+         * @param[in] serialnumber NACA-serial number for different airfoil
+         * shapes (so far only 4-digit-series implemented).
+         * @param[in] number_points Defines the amount of points for each side.
+         * @param[in] is_upper Bool to choose either upper or lower side.
+         * @return naca_create_points_4_digits Vector containing
+         *  (number_points)-upper or lower airfoil points (non-equidistant).
+         */
+        static std::vector<Point<2>>
+        naca_create_points(const std::string &serialnumber,
+                           const unsigned int number_points,
+                           const bool         is_upper)
+        {
+          Assert(serialnumber.length() == 4,
+                 ExcMessage("This NACA-serial number is not implemented!"));
+
+          return naca_create_points_4_digits(serialnumber,
+                                             number_points,
+                                             is_upper);
+        }
+
+        /**
+         * Calculate airfoil points for 4-digit NACA airfoils according to
+         * following reference literature
+         *
+         * I.H. Abbott and A.E. von Doenhoff. Theory of Wing Sections: Including
+         * a Summary of Airfoil Data. New York: Dover Publications, 1959.
+         *
+         * @param[in] serialnumber NACA-serialnumber for different airfoilshapes
+         *  (so far only 4-digit-series implemented).
+         * @param[in] number_points Defines the amount of points for each side.
+         * @param[in] is_upper Bool to choose either upper or lower side.
+         * @return naca_points Vector containing (number_points)-upper or lower
+         *  airfoil points (non-equidistant).
+         */
+        static std::vector<Point<2>>
+        naca_create_points_4_digits(const std::string &serialnumber,
+                                    const unsigned int number_points,
+                                    const bool         is_upper)
+        {
+          // conversion string (char * ) to int
+          const unsigned int digit_0 = int(serialnumber[0]) - 48;
```

Comment:
I think this might be easier to read if you wrote it like this:
```suggestion
          const unsigned int digit_0 = (serialnumber[0] - '0');
```
which I believe is the same.